oracle/types/oracle.go

```diff
@@ -14,7 +14,6 @@ import (
 	wsmetrics "github.com/skip-mev/slinky/providers/base/websocket/metrics"
 	providertypes "github.com/skip-mev/slinky/providers/types"
 	"github.com/skip-mev/slinky/providers/types/factory"
-	mmtypes "github.com/skip-mev/slinky/x/marketmap/types"
 )
 
```

Comment:
Note, every single alias that is relevant to a provider will now utilize the ProviderTicker interface as oppose to mmtypes.Ticker. The aggregator now maps (provider, off-chain-ticker) -> price. This will make it much easier to index prices from the aggregator with a specified market map.

Reply:
I think the following is true...it's been hard to reason about things because of the rewrites going on. Let me know if this makes sense or is completely wrong.

This means we now have 2 uniqueness constraints right? `mmtypes.Ticker`--this is what we would call a "canonical ticker" or something like that which is `base/quote`. This has to be unique across all Markets and that is enforced in the keeper.

And now we have `off-chain-ticker` which is similar, but only has to be unique per Provider. That's not really enforced anywhere. It's checked in `ProviderTickersFromMarketMap`. If we hit an offchain ticker we've already seen, we ignore it. Right now the only thing that could be different between them is the json metadata so it's probably not a big deal, but it would result in indeterminism based on iteration order.

Reply:
Yes, there should be a unique-ness check based on the off-chain-ticker and the json metadata. Our assumption atm is that In no case should the same provider have the same off-chain-ticker but have different json meta-data.

Reply:
we should probably add that as a check in the module.